In [35]:
from keras.datasets import cifar10
# loading the dataset 
(X_train, y_train), (X_test, y_test) = cifar10.load_data()

# keras imports for the dataset and building our neural network


In [36]:
from keras.datasets import cifar10
from keras.models import Sequential
from keras.layers import Dense, Dropout, Conv2D, MaxPool2D, Flatten
from keras.utils import np_utils

# loading the dataset



In [37]:
(X_train, y_train), (X_test, y_test) = cifar10.load_data()

#  building the input vector from the 32x32 pixels

In [38]:
X_train = X_train.reshape(X_train.shape[0], 32, 32, 3)
X_test = X_test.reshape(X_test.shape[0], 32, 32, 3)
X_train = X_train.astype('float32')
X_test = X_test.astype('float32')


# normalizing the data to help with the training

In [39]:

X_train /= 255
X_test /= 255

# one-hot encoding using keras' numpy-related utilities



In [40]:
n_classes = 10
print("Shape before one-hot encoding: ", y_train.shape)
Y_train = np_utils.to_categorical(y_train, n_classes)
Y_test = np_utils.to_categorical(y_test, n_classes)
print("Shape after one-hot encoding: ", Y_train.shape)

Shape before one-hot encoding:  (50000, 1)
Shape after one-hot encoding:  (50000, 10)


# building a linear stack of layers with the sequential model

In [41]:
model = Sequential()

# convolutional layer

In [42]:
model.add(Conv2D(50, kernel_size=(3,3), strides=(1,1), padding='same', activation='relu', input_shape=(32, 32, 3)))


#  Adding the convolutional layer

In [43]:
model.add(Conv2D(75, kernel_size=(3,3), strides=(1,1), padding='same', activation='relu'))
model.add(MaxPool2D(pool_size=(2,2)))
model.add(Dropout(0.25))


In [44]:
model.add(Conv2D(125, kernel_size=(3,3), strides=(1,1), padding='same', activation='relu'))
model.add(MaxPool2D(pool_size=(2,2)))
model.add(Dropout(0.25))

#adding one more layer of one convolution layer

In [45]:
model.add(Conv2D(150, kernel_size=(3,3), strides=(1,1), padding='same', activation='relu'))
model.add(MaxPool2D(pool_size=(2,2)))
model.add(Dropout(0.25))

# flatten output of conv


In [46]:
model.add(Flatten())

# hidden layer


In [47]:
model.add(Dense(500, activation='relu'))
model.add(Dropout(0.4))
model.add(Dense(250, activation='relu'))
model.add(Dropout(0.3))

when i add one more hidden layer

In [49]:
model.add(Dense(300, activation='relu'))
model.add(Dropout(0.5))

# output layer



In [50]:
model.add(Dense(10, activation='softmax'))

# compiling the sequential model


In [51]:
model.compile(loss='categorical_crossentropy', metrics=['accuracy'], optimizer='adam')

# training the model for 10 epochs

In [52]:
model.fit(X_train, Y_train, batch_size=128, epochs=10, validation_data=(X_test, Y_test))

Epoch 1/10
391/391 [==============================] - 9s 22ms/step - loss: 1.7992 - accuracy: 0.3177 - val_loss: 1.3508 - val_accuracy: 0.5012
Epoch 2/10
391/391 [==============================] - 9s 22ms/step - loss: 1.2917 - accuracy: 0.5328 - val_loss: 1.0718 - val_accuracy: 0.6202
Epoch 3/10
391/391 [==============================] - 8s 22ms/step - loss: 1.0837 - accuracy: 0.6181 - val_loss: 0.8893 - val_accuracy: 0.6910
Epoch 4/10
391/391 [==============================] - 9s 22ms/step - loss: 0.9528 - accuracy: 0.6702 - val_loss: 0.8205 - val_accuracy: 0.7140
Epoch 5/10
391/391 [==============================] - 9s 22ms/step - loss: 0.8723 - accuracy: 0.6977 - val_loss: 0.7719 - val_accuracy: 0.7331
Epoch 6/10
391/391 [==============================] - 8s 22ms/step - loss: 0.8096 - accuracy: 0.7211 - val_loss: 0.7353 - val_accuracy: 0.7523
Epoch 7/10
391/391 [==============================] - 8s 22ms/step - loss: 0.7529 - accuracy: 0.7407 - val_loss: 0.7122 - val_accuracy: 0.7553

**Note**
1. When I have add 2 convolution layer than i get the accuracy of 77 % 
2. when I have added one more convolution layer than i get the accuracy of 80 %
3. There is no any improvement in accuracy while adding one more hiddden layer